Source/WebInspectorUI/UserInterface/Base/Main.js

```diff
@@ -2098,393 +2087,18 @@ WI._toggleInspectMode = function(event)
-                return;
-
-            if (!force && (!value || value === "default")) {
-                target.PageAgent.setScreenSizeOverride((error) => {
-                    if (error) {
-                        WI.reportInternalError(error);
-                        return;
-                    }
-
-                    WI._overridenDeviceScreenSize = null;
-                    updateActivatedState();
-                    target.PageAgent.reload();
-                });
-            } else {
-                let tokens = value.split("x");
-                let width = parseInt(tokens[0]);
-                let height = parseInt(tokens[1]);
-                target.PageAgent.setScreenSizeOverride(width, height, (error) => {
-                    if (error) {
-                        WI.reportInternalError(error);
-                        return;
-                    }
-
-                    WI._overridenDeviceScreenSize = value;
-                    updateActivatedState();
-                    target.PageAgent.reload();
-                });
-            }
-        }
-
-
-        let screenSizeRow = table.appendChild(document.createElement("tr"));
-
-        let screenSizeTitle = screenSizeRow.appendChild(document.createElement("td"));
-        screenSizeTitle.textContent = WI.UIString("Screen size:");
-
-        let screenSizeValue = screenSizeRow.appendChild(document.createElement("td"));
-        screenSizeValue.classList.add("screen-size");
-
-        let screenSizeValueSelect = screenSizeValue.appendChild(document.createElement("select"));
-
-        let screenSizeValueInput = null;
-
-        const screenSizes = [
-            [
-                {name: WI.UIString("Default"), value: "default"},
-            ],
-            [
-                {name: WI.UIString("1080p"), value: "1920x1080"},
-                {name: WI.UIString("720p"), value: "1280x720"},
-            ],
-            [
-                {name: WI.UIString("Other\u2026"), value: "other"},
-            ],
-        ];
-
-        let selectedScreenSizeOptionElement = null;
-
-        for (let group of screenSizes) {
-            for (let {name, value} of group) {
-                let optionElement = screenSizeValueSelect.appendChild(document.createElement("option"));
-                optionElement.value = value;
-                optionElement.textContent = name;
-
-                if (value === WI._overridenDeviceScreenSize)
-                    selectedScreenSizeOptionElement = optionElement;
-            }
-
-            if (group !== screenSizes.lastValue)
-                screenSizeValueSelect.appendChild(document.createElement("hr"));
-        }
-
-        function showScreenSizeInput() {
-            if (screenSizeValueInput)
-                return;
+    if (WI._deviceSettingsPopover)
+        return; // Clicking the button while the popover is shown will automatically dismiss the popover. The object is cleared by `WI.didDismissPopover()`.
```

Comment:
NIT: I kinda feel like we should make it so that clicking the button either dismisses the popover without showing it again (just like what color swatches do) or doesn't dismiss it at all 🤔

Reply:
Isn't that what this line achieves? Since a click occurs outside the popover, it should clean itself up, and this early return should prevent the popover from presenting itself? Or am I mistaken @rcaliman-apple?

Reply:
`InlineSwatch` does the same thing. It's just handled internally: [InlineSwatch.js#L144-L170](https://github.com/WebKit/WebKit/blob/main/Source/WebInspectorUI/UserInterface/Views/InlineSwatch.js#L144-L170) and uses `_valueEditor` as the object instance to [check for](https://github.com/WebKit/WebKit/blob/main/Source/WebInspectorUI/UserInterface/Views/InlineSwatch.js#L277-L278) and [nullify](https://github.com/WebKit/WebKit/blob/main/Source/WebInspectorUI/UserInterface/Views/InlineSwatch.js#L167).